### ISSUE TITLE
Android build issue: Resource.Layout does not contain a definition for 'Tabbar'

### ISSUE THREAD

**Issue by jhakulin** [xamarin/xamarin-android#5088]
### Steps to Reproduce

1.  Build  https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/xamarin solution for Android

### Expected Behavior
Build pass without issues

### Actual Behavior
Occasionally in our CI environment, there is build failure which is caused by:
 
2020-09-04T20:17:58.4341434Z          Using shared compilation with compiler from directory: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn
2020-09-04T20:18:00.0289772Z ##[error]public_samples\quickstart\csharp\xamarin\helloworld\helloworld.Android\MainActivity.cs(26,49): Error CS0117: 'Resource.Layout' does not contain a definition for 'Tabbar'
2020-09-04T20:18:00.0300060Z      3>MainActivity.cs(26,49): error CS0117: 'Resource.Layout' does not contain a definition for 'Tabbar' [D:\a\1\s\public_samples\quickstart\csharp\xamarin\helloworld\helloworld.Android\helloworld.Android.csproj]
2020-09-04T20:18:00.0310531Z ##[error]public_samples\quickstart\csharp\xamarin\helloworld\helloworld.Android\MainActivity.cs(27,47): Error CS0117: 'Resource.Layout' does not contain a definition for 'Toolbar'
2020-09-04T20:18:00.0327854Z      3>MainActivity.cs(27,47): error CS0117: 'Resource.Layout' does not contain a definition for 'Toolbar' [D:\a\1\s\public_samples\quickstart\csharp\xamarin\helloworld\helloworld.Android\helloworld.Android.csproj]
2020-09-04T20:18:00.0382584Z      3>Done Building Project "D:\a\1\s\public_samples\quickstart\csharp\xamarin\helloworld\helloworld.Android\helloworld.Android.csproj" (default targets) -- FAILED.
2020-09-04T20:18:00.0393819Z      1>Done Building Project "D:\a\1\s\public_samples\quickstart\csharp\xamarin\helloworld.sln" (default targets) -- FAILED.

### Version Information
Visual Studio 2019

### Workaround
When rerun CI build, solution occasionally builds fine.
We have also updated quickstart to latest VS2019 Xamarin C# template project, that seemed to help at the beginning but problem happens still occasionally

Is there a workaround for this problem ?

**Comment by jpobst** [comment]
This sounds like https://github.com/xamarin/xamarin-android/pull/5107.

Can you try setting it to build 1 project at a time and see if that fixes it?

- Tools
- Options
- Projects and Solutions
- Build and Run
- Maximum number of parallel project builds = 1.

**Comment by jpobst** [comment]
I just noticed you said CI, for `msbuild` it's:

```
msbuild -m:1
```

**Comment by jonathanpeppers** [comment]
@jhakulin let us know if you are using `-m` at all, I think you can just drop it from the MSBuild command completely.

That would help us know if #5107 will fix your problem, thanks.

**Comment by jhakulin** [comment]
@jonathanpeppers Thanks for information, I will check that.

**Comment by EmilAlipiev** [comment]
> This sounds like #5107.
> 
> Can you try setting it to build 1 project at a time and see if that fixes it?
> 
> * Tools
> * Options
> * Projects and Solutions
> * Build and Run
> * Maximum number of parallel project builds = 1.

your suggestions solves my problem. It seems like this my issue is related. https://github.com/xamarin/xamarin-android/issues/5141

For me it only happens when I use an xamarin android library project type. Strange that it works with Rider. Do they use different Xamarin.Android than VS2019?

**Comment by jonathanpeppers** [comment]
The bug fixed in #5141 is caused by MSBuild running on multiple nodes, if you have more than 1 listed here:

![image](https://user-images.githubusercontent.com/840039/93825135-45e9d600-fc2a-11ea-90b3-74e4fedb8ae8.png)

I suspect Rider just uses a single MSBuild node, and that is why it works?

**Comment by EmilAlipiev** [comment]
I think there is major issue with xamarin.android builds it seems to be VS related because as I said i have no problem using Rider and when I want to test my multitargeting nuget package sourcecode together with demo projects, I get every time below error message. Deleting bin and obj solves the problem but it is every single time even doing "rebuild".  

` System.IO.IOException: The directory is not empty.`

**Comment by jonathanpeppers** [comment]
>  test my multitargeting nuget package sourcecode together with demo projects...
> `System.IO.IOException: The directory is not empty.`

@EmilAlipiev can you file a new issue with [diagnostic MSBuild logs](https://docs.microsoft.com/xamarin/android/troubleshooting/troubleshooting#diagnostic-msbuild-output)? This sounds like something different than the issue posted here.

**Comment by bradgrace** [comment]
@jpobst I ran a very similar, if not the same issue, where a definition for `Tabbar` wasn't found:

```
error CS0117: 'Resource.Layout' does not contain a definition for 'Tabbar'
error CS0117: 'Resource.Layout' does not contain a definition for 'Toolbar'
error CS0117: 'Resource.Drawable' does not contain a definition for 'enabled'
error CS0117: 'Resource.Drawable' does not contain a definition for 'disabled'
```

In my case, I was building within VS 2019. I followed the advice mentioned in https://github.com/xamarin/xamarin-android/issues/5088#issuecomment-692281407 and that seems to have resolved it. It seems to require cleaning the project and then building twice before the issue is resolved. Which is a bit odd, but what I've observed. After that, the build succeeds each time. 

And, after switching the parallel value back to a value greater than 1, the build immediately fails. So, I feel confident that your suggestion provides a valid workaround.
